.github/workflows/go-mod.yaml

```
@@ -0,0 +1,22 @@
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - 'v**'
+
+name: Go Mod Validation
+jobs:
+  go-mod-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+
+      - run: go mod tidy
+      - run: go mod vendor
+      - run: |
+          if [ -n "$(git status --porcelain --untracked-files=no)" ]; then
+              echo "go.mod and/or go.sum are out of date, please run `go mod tidy && go mod vendor` and update your changes"
+              exit 1
```

```suggestion
              git diff
              exit 1
```

would be useful to see the changes!

---

thanks, ill add it!